crm_salesperson_planner/models/crm_salesperson_planner_visit.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2021 Sygel - Valentin Vinagre
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class CrmSalespersonPlannerVisit(models.Model):
+    _name = "crm.salesperson.planner.visit"
+    _description = "Salesperson Planner Visit"
+    _order = "date desc,sequence"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+
+    name = fields.Char(
+        string="Visit Number", required=True, default="/", readonly=True, copy=False,
+    )
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Customer", required=True,
+    )
+    partner_phone = fields.Char(string="Phone", related="partner_id.phone")
+    partner_mobile = fields.Char(string="Mobile", related="partner_id.mobile")
+    date = fields.Date(string="Date", default=fields.Date.context_today, required=True,)
+    sequence = fields.Integer(
+        string="Sequence",
+        help="Used to order Visits in the different views",
+        default=20,
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        default=lambda self: self.env.company,
+    )
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Salesperson",
+        index=True,
+        tracking=True,
+        default=lambda self: self.env.user,
+        domain=lambda self: [
+            ("groups_id", "in", self.env.ref("sales_team.group_sale_salesman").id)
+        ],
+    )
+    opportunity_ids = fields.Many2many(
+        comodel_name="crm.lead",
+        relation="crm_salesperson_planner_visit_crm_lead_rel",
+        string="Opportunities",
+        copy=False,
+        domain="[('type', '=', 'opportunity'), ('partner_id', 'child_of', partner_id)]",
+    )
+    description = fields.Html(string="Description")
+    state = fields.Selection(
+        string="Status",
+        required=True,
+        readonly=True,
+        copy=False,
+        tracking=True,
+        selection=[
+            ("draft", "Draft"),
+            ("confirm", "Validated"),
+            ("done", "Visited"),
+            ("cancel", "Cancelled"),
+            ("incident", "Incident"),
+        ],
+        default="draft",
+    )
+    close_reason_id = fields.Many2one(
+        comodel_name="crm.salesperson.planner.visit.close.reason", string="Close Reason"
+    )
+    close_reason_image = fields.Image(
+        string="Close Reason Image", max_width=1024, max_height=1024
+    )
+    clase_reason_notes = fields.Text(string="Close Reason Notes")
+    visit_template_id = fields.Many2one(
+        comodel_name="crm.salesperson.planner.visit.template", string="Visit Template"
+    )
+
+    _sql_constraints = [
+        (
+            "crm_salesperson_planner_visit_name",
+            "UNIQUE (name)",
+            "The name must be unique!",
```

Comment:
```suggestion
            "The visit number must be unique!",
```

but it's interesting to have another code more? Isn't the unique should come by partner_id and date?